### PR TITLE
Feature/sl 962/sl 984 ui dialog

### PR DESCRIPTION
**Pull request by leszekgu**
Have a look and tell me what you think.

### CODE REVIEW

File: src/app/sessions/components/session-edit-or-create-dialog/session-edit-or-create-dialog.component.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        this.sessionEditData = {
+            editedSession: this.sessionDetails,
+            onCancel: this.cancel,
+            dialogReference: this.thisDialogRef
+        } as SessionEditData;
+    }
+
+    cancel(event) {
+        event.details.dialogReference.close();
+        console.log('cancel clicked', event);
```
Comment (zielinskikamil) [inline src/app/sessions/components/session-edit-or-create-dialog/session-edit-or-create-dialog.component.ts]: do we need this console.log here ? :)

Comment (leszekgu) [reply src/app/sessions/components/session-edit-or-create-dialog/session-edit-or-create-dialog.component.ts]: lol, no, but that reminded me that I should remove a bit more from here.

File: src/app/sessions/models/session-edit-data.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { SessionCreate } from './session-create.model';
+import { SessionEditOrCreateDialogComponent } from '../components/session-edit-or-create-dialog/session-edit-or-create-dialog.component';
+import { MatDialogRef } from '@angular/material';
+
+export interface SessionEditData {
```
Comment (radoslawlandowskikainos) [inline src/app/sessions/models/session-edit-data.ts]: SessionEditData contains a callback and dialog reference. Seems like stuff got mixed. I would not include the onCancel and DialogReference in this model which would leave it only with SessionCreate, which then makes SessionEditData not needed

File: src/app/sessions/components/session-edit-or-create-dialog/session-edit-or-create-dialog.component.html [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<div class="sessions-create-container">
```
Comment (radoslawlandowskikainos) [inline src/app/sessions/components/session-edit-or-create-dialog/session-edit-or-create-dialog.component.html]: What's the purpose of having such component? It looks like a carrier of data and dialog wrapper. I can see it sets the reloadDataOnInit flag to false. Is there anything else? Could we get rid of it? Besides, Why 'edit-or-create' ? You cannot edit a session with it, you can only create a new one

File: src/app/sessions/components/sessions-create-form/sessions-create-form.component.ts [hunk 4]
```diff
@@ -4,60 +4,87 @@ import { Room } from '../../../rooms/models/room.model';
...
-  }
+    session: SessionCreate;
+    durationInMinutes: number;
+    caseTypes: string[];
+    time: string;
+    roomsPlaceholder: string;
+    judgesPlaceholder: string;
 
+    private _sessionEditData: SessionEditData;
+    @Input() set sessionEditData(value: SessionEditData) {
```
Comment (radoslawlandowskikainos) [inline src/app/sessions/components/sessions-create-form/sessions-create-form.component.ts]: Why name 'sessionEditData'? There seem to be no editing to that component. It seems like name _initSessionData would work better for the purpose

File: src/app/sessions/containers/sessions-propositions-search/sessions-propositions-search.component.ts [hunk 5]
```diff
@@ -22,6 +26,7 @@ import { combineLatest } from 'rxjs/observable/combineLatest';
 })
 export class SessionsPropositionsSearchComponent implements OnInit {
```
Comment (radoslawlandowskikainos) [inline src/app/sessions/containers/sessions-propositions-search/sessions-propositions-search.component.ts]: SessionsPropositionsSearchComponent is our smart container here and has all the data that CreateSessionForm requires. Why do we then need to use SessionsCreateComponent (another smart container) inside  SessionEditOrCreateDialogComponent?

now we have: smart -> dumb -> smart -> dumb
why not simply: smart(SessionsPropositionsSearchComponent) -> dumb(SessionsCreateFormComponent) ?

**Pull request merged**